bi_view_editor/models/ir_model.py

```diff
@@ -0,0 +1,282 @@
+                                 table_alias=model[0])
+                        )
+            return model_list
+
+        def get_relation_list(model_ids, model_names):
+            if not model_ids:
+                return []
+            self.env.cr.execute('''
+                SELECT imf.id, imf.model_id, imf.name,
+                       imf.field_description AS description,
+                       imf.ttype AS type,
+                       relation, imf.model,
+                       im.name AS model_name
+                FROM ir_model_fields imf
+                    JOIN ir_model im ON im.id = imf.model_id
+                WHERE imf.relation IN %s
+                    AND imf.store = true
+                    AND imf.ttype = 'many2one'
+                ORDER BY field_description DESC''', (
+                tuple(model_names.values()),))
+            filtered_fields = self.env.cr.dictfetchall()
+            relation_list = []
+            for model in model_ids.items():
+                for field in filtered_fields:
+                    if model_names[model[1]] == field['relation']:
+                        relation_list.append(
+                            dict(field,
+                                 join_node=model[0],
+                                 table_alias=-1)
+                        )
+            return relation_list
+
+        models = self.sudo().browse(model_ids.values())
+        model_names = {}
+        for model in models:
+            model_names.update({model.id: model.model})
+
+        model_list = get_model_list(model_ids)
+        relation_list = get_relation_list(model_ids, model_names)
+
+        return relation_list + model_list
+
+    @api.model
+    def get_related_models(self, model_ids):
+        """ Return list of model dicts for all models that can be
+            joined with the already selected models.
+        """
+
+        def _get_field(fields, orig, target):
+            field_list = []
+            for f in fields:
+                if f[orig] == -1:
+                    field_list.append(f[target])
+            return field_list
+
+        def _get_list_id(model_ids, fields):
+            list_model = list(model_ids.values())
+            list_model += _get_field(fields, 'table_alias', 'model_id')
+            return list_model
+
+        def _get_list_relation(fields):
+            list_model = _get_field(fields, 'join_node', 'relation')
+            return list_model
+
+        domain = [('transient', '=', False)]
+        if model_ids:
+            related_fields = self.get_related_fields(model_ids)
+            list_id = _get_list_id(model_ids, related_fields)
+            list_model = _get_list_relation(related_fields)
+            domain += ['|',
+                       ('id', 'in', list_id),
+                       ('model', 'in', list_model)]
+        return self.sudo().search(domain)
+
+    @api.model
+    def get_models(self, model_ids=None):
```

Comment:
`model_ids={}` instead

Reply:
This proposed change causes a warning:

```
************* Module bi_view_editor.models.ir_model
bi_view_editor/models/ir_model.py:173: [W0102(dangerous-default-value), IrModel.get_models] Dangerous default value {} as argument

```

so I will revert it back to `model_ids=None`.